app/src/main/java/pl/org/akai/eater/viewmodel/AuthViewModel.kt

```diff
@@ -0,0 +1,24 @@
+package pl.org.akai.eater.viewmodel
+
+import android.util.Log
+import androidx.lifecycle.ViewModel
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import pl.org.akai.eater.base.BaseViewModel
+import pl.org.akai.eater.base.UiEffect
+import pl.org.akai.eater.base.UiEvent
+import pl.org.akai.eater.base.UiState
+import pl.org.akai.eater.util.User
+
+class AuthViewModel : ViewModel() {
+    private val _user: MutableStateFlow<User?> = MutableStateFlow(null)
+    val user: StateFlow<User?> = _user
```

Comment:
It is small viewModel, but it would be much better to have State class which will contain User class